File: src/UI/Mobile/FlutterApp/lib/features/audit/data/audit_service.dart
import '../../../core/network/api_client.dart';

class AuditService {
  final ApiClient _client;

  AuditService(this._client);

  Future<dynamic> getAuditLogs({required String entityType, int page = 1, int pageSize = 10}) async {
    final response = await _client.dio.get(
      '/api/v1/audit/$entityType',
      queryParameters: {'page': page, 'pageSize': pageSize},
    );
    return response.data;
  }
}

File: src/UI/Web/React/src/api/services/audit.service.ts
import { BaseService } from '../base.service';
import { AuditLogResponse } from '../../types';

export class AuditService extends BaseService<AuditLogResponse> {
  constructor() {
    super('/api/v1/audit');
  }

  /**
   * Sobrescreve o list para lidar com o endpoint específico por tipo de entidade
   */
  async getAuditLogs(page = 1, pageSize = 10, filters: { entityType?: string; eventType?: string; userId?: string; from?: string; toDate?: string } = {}) {
    const { entityType, ...otherFilters } = filters;
    const path = entityType ? `${this.entityPath}/${entityType}` : this.entityPath;
    const query = this.toQueryString({ page, pageSize, ...otherFilters });
    
    const response = await this.http.get(path + query);
    return response.data;
  }
}

export const auditService = new AuditService();
